ledger_device_sdk/src/hash.rs

```diff
@@ -67,7 +67,7 @@ pub trait HashInit: Sized {
             Ok(())
         }
     }
-    fn finalize(mut self, output: &mut [u8]) -> Result<(), HashError> {
+    fn finalize(&mut self, output: &mut [u8]) -> Result<(), HashError> {
         let output_size = self.get_size();
```

Comment:
I do not think this is required:when a hash is finalized, then its context shall be consumed. To compute a new hash, then `HashInit::new()` shall be called. It is the same behaviour as with `HashInit::hash()`